linchpin_api/v1/api.py

```diff
@@ -0,0 +1,214 @@
+import os
+import inspect
+import ansible
+import pprint
+import linchpin_api
+from tabulate import tabulate
+from ansible import utils
+import jsonschema as jsch
+from collections import namedtuple
+from ansible import utils
+from ansible.parsing.dataloader import DataLoader
+from ansible.vars import VariableManager
+from ansible.inventory import Inventory
+from ansible.executor.playbook_executor import PlaybookExecutor
+from ansible.plugins.callback import CallbackBase
+from callbacks import PlaybookCallback
+from invoke_playbooks import invoke_linchpin
+from utils import get_file, list_files, parse_yaml
+from github import GitHub
+
+
+class LinchpinAPI:
+
+    def __init__(self):
+        self.PLAYBOOKS = {
+                           "PROVISION": "site.yml",
+                           "TEARDOWN": "site.yml",
+                           "SCHEMA_CHECK": "schema_check.yml",
+                           "INVGEN": "invgen.yml",
+                         }
+        base_path = os.path.dirname(__file__).split("/")
+        base_path = base_path[0:-2]
+        self.base_path = "/".join(base_path)
+        self.excludes = set(["topology_upstream",
+                             "layout_upstream",
+                             "post_actions"])
+
+    def get_config_path(self):
+        try:
+            config_files = [
+                            "./linchpin_config.yml",
+                            "~/.linchpin_config.yml",
+                            self.base_path+"/linchpin_config.yml",
+                            "/etc/linchpin_config.yml"]
+            for c_file in config_files:
+                if os.path.isfile(c_file):
+                    return c_file
+        except Exception as e:
+            print(e)
+
+    def get_config(self):
+        config_path = self.get_config_path()
+        config = parse_yaml(config_path)
+        return config
+
+    def get_evars(self, lpf):
+        """ creates a group of extra vars on basis on linchpin file dict """
+        e_vars = []
+        for group in lpf:
+            topology = lpf[group].get("topology")
+            layout = lpf[group].get("layout")
+            e_var_grp = {}
+            e_var_grp["topology"] = search_path(topology, os.getcwd())
+            e_var_grp["layout"] = search_path(layout, os.getcwd())
+            if None in e_var_grp.values():
+                display("ERROR:003")
```

Comment:
It would seem an ambiguous error to say one or the other cannot be found. Instead of doing this, a try/catch would be more pythonic. Consider the following:

    def find_file(self, topology, path):
       topo = search_path(topology, path)
       if topo is None:
            raise FileNotFoundError(
                     'topology {} could not be found at path {}'.format(topology, path))
       return topo

  ..snip..

      try:
        e_var_grp["topology"] = find_file(topology, os.getcwd())
    except FileNotFoundError e:
        raise LinchPinError(e)

This not only handles exceptions better, it makes the code more readable. Additionally, doing this would give the added benefit of providing an optional search path in the Pinfile